src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -298,12 +212,54 @@ export class MagicWandService {
         }
         // Visits the pixel and check if it should be part of the mask.
         visited.add(this.coordToDataArrayIndex(x, y, imgData.width));
-        if (this.getIsScribbleMask(scribbles, imgData, neighborPixel, tolerance)) {
+        // if (this.getIsScribbleMask(scribbles, imgData, neighborPixel, tolerance)) {
+        //   visit.push(neighborPixel);
+        // }
+        if (tree.nearest(
+            this.scribblesToColors(new Set([
+            this.coordToDataArrayIndex(x, y, imgData.width)]), imgData)[0],
+            1, tolerance * tolerance).length > 0) {
           visit.push(neighborPixel);
         }
       }
     }  // End of while loop.
     
     return mask;
   }
+
+  /**@returns {number} the straight line distance between the two colors
+   * @param {Array<number> [R, G, B]} basisColor and 
+   * @param {Array<number> [R, G, B]} secondColor
+   */
+  rgbEuclideanDist(colorA: Color, colorB: Color): number {
+
+    let distance = 0;
+
+    for (let attribute in colorA) {
+      distance += Math.pow((colorA[attribute] - colorB[attribute]), 2);
+    }
+
+    // Does not sqrt distance to complete eucidean dist formula b/c sqrt is
+    // an expense operation. Instead, can compare against tolerance in the 
+    // squared space.
+    return distance;
+  }
+
+  /**@license MIT License <http://www.opensource.org/licenses/mit-license.php>
+   * Pretty good color distance from
+   * http://www.compuphase.com/cmetric.htm
+   */
+  colorDistance(a: Color, b: Color): number {
+    var dr = a.red - b.red;
```

Comment:
not sure if this applies for typescript/angular, but https://google.github.io/styleguide/jsguide.html#features-use-const-and-let mentions using const or let over var.